pkg/controller/master/node/cpu_manager_controller.go

```diff
@@ -0,0 +1,322 @@
+	CPUManagerNonePolicy   CPUManagerPolicy = "none"
+	// status
+	CPUManagerRequestedStatus CPUManagerStatus = "requested"
+	CPUManagerRunningStatus   CPUManagerStatus = "running"
+	CPUManagerSuccessStatus   CPUManagerStatus = "success"
+	CPUManagerFailedStatus    CPUManagerStatus = "failed"
+
+	CPUManagerRootMountPath         string = "/host"
+	CPUManagerScriptMountPath       string = "/harvester-helpers"
+	CPUManagerScriptPath            string = CPUManagerScriptMountPath + "/cpu-manager.sh"
+	CPUManagerWaitLabelTimeoutInSec int64  = 300 // 5 min
+	CPUManagerJobTimeoutInSec       int64  = CPUManagerWaitLabelTimeoutInSec * 2
+	CPUManagerJobTTLInSec           int32  = 86400 * 7 // 7 day
+)
+
+type CPUManagerPolicy string
+type CPUManagerStatus string
+
+type CPUManagerUpdateStatus struct {
+	Policy  CPUManagerPolicy `json:"policy"`
+	Status  CPUManagerStatus `json:"status"`
+	JobName string           `json:"jobName,omitempty"`
+}
+
+// cpuManagerNodeHandler updates cpu manager status of a node in its annotations, so that
+// we can tell whether the node is under modifing cpu manager policy or not and its current policy.
+type cpuManagerNodeHandler struct {
+	appCache   catalogv1.AppCache
+	nodeCache  ctlcorev1.NodeCache
+	nodeClient ctlcorev1.NodeClient
+	jobClient  ctlbatchv1.JobClient
+	namespace  string
+}
+
+// CPUManagerRegister registers the node controller
+func CPUManagerRegister(ctx context.Context, management *config.Management, options config.Options) error {
+	app := management.CatalogFactory.Catalog().V1().App()
+	job := management.BatchFactory.Batch().V1().Job()
+	node := management.CoreFactory.Core().V1().Node()
+
+	cpuManagerNodeHandler := &cpuManagerNodeHandler{
+		appCache:   app.Cache(),
+		jobClient:  job,
+		nodeCache:  node.Cache(),
+		nodeClient: node,
+		namespace:  options.Namespace,
+	}
+
+	node.OnChange(ctx, CPUManagerControllerName, cpuManagerNodeHandler.OnNodeChanged)
+	job.OnChange(ctx, CPUManagerControllerName, cpuManagerNodeHandler.OnJobChanged)
+
+	return nil
+}
+
+func (h *cpuManagerNodeHandler) OnNodeChanged(_ string, node *corev1.Node) (*corev1.Node, error) {
+	if node == nil || node.Annotations == nil || node.DeletionTimestamp != nil {
+		return node, nil
+	}
+
+	annot, found := node.Annotations[util.AnnotationCPUManagerUpdateStatus]
+	if !found {
+		return node, nil
+	}
+
+	// do nothing if the node is in witness role
+	if _, found := node.Labels[HarvesterWitnessNodeLabelKey]; found {
+		return node, nil
+	}
+
+	cpuManagerStatus, err := GetCPUManagerUpdateStatus(annot)
+	if err != nil {
+		logrus.WithField("node_name", node.Name).WithError(err).Error("Skip update cpu manager policy, failed to retrieve cpu-manager-update-status from annotation")
+		return node, nil
+	}
+	// do nothing if status is not requested
+	if cpuManagerStatus.Status != CPUManagerRequestedStatus {
+		return node, nil
+	}
+
+	newNode := node.DeepCopy()
+	job, err := h.submitJob(cpuManagerStatus, node)
```

Comment:
There are scenarios: 

`submitJob` is successful but `h.nodeClient.Update` failed (e.g. node is updated by other controller on the meanwhile)

following code will re-create the job;  LB code ever hit similar issue and caused IP duplicated allocation

this is a general issue when a controller updates several CRD objects

because the job is created with generated name, there is no elegant solution:
either before submitJob, check if the label matched job has been there;
or in the job itself, check if a peer is running

```
	job, err := h.submitJob(cpuManagerStatus, node)
...
	return h.nodeClient.Update(newNode)
	```